collector/kernel/kernel_collector.cc

```diff
@@ -660,6 +677,10 @@ void KernelCollector::enter_try_connecting(std::chrono::milliseconds discount)
     timeout += std::chrono::milliseconds{d(rd)};
   }
 
+  if (kernel_collector_restarter_ && kernel_collector_restarter_->restart_in_progress_) {
```

Comment:
We're doing a bunch of work, including an expensive clock call and consuming from the entropy pool above.. none of which is needed when this condition (known in advance) is true.  Perhaps we should conditionally wrap the entire expected behavior above - as opposed to doing the work, and then throwing it out?

Reply:
Throwing out the work should be a very rare occurrence, but I'll take a look at refactoring.

Reply:
Refactored.